packages/frameworks-astro/src/components/SignIn.astro

```diff
@@ -0,0 +1,32 @@
+---
+import type { HTMLAttributes } from 'astro/types'
+import type { BuiltInProviderType } from '@auth/core/providers';
+import type { LiteralUnion, SignInAuthorizationParams, SignInOptions } from 'next-auth/react/types';
+
+interface Props extends HTMLAttributes<'button'> {
+  /** The authentication provider to sign in with. */
+  provider?: LiteralUnion<BuiltInProviderType, string>
+  options?: SignInOptions
+  authParams?: SignInAuthorizationParams
+}
+
+const key = Math.random().toString(36).slice(2, 11)
+
+const { provider, options, authParams, ...attrs } = Astro.props
+attrs.class = `signin-${key} ${attrs.class ?? ''}`
```

Comment:
tiny nit: I usually recommend leaning on `class:list` when passing `class` props to Astro components

```html
---
const { class: className, ...attrs } = Astro.props
---

<button class:list={[`signin-${key}`, className]} {...attrs}>
    <slot />
</button> 
```